### PR TITLE
fix(material/checkbox): fix ARIA semantics and use native DOM properties

### LINKED ISSUE #26709: bug(material/checkbox): when both checked and indeterminate Inputs are true, checkbox displays indeterminate state but a11y tree has state of "checked"
### Is this a regression?

- Look slike this regression was introduced in #8089 

### The previous version in which this bug was not present was

_No response_

### Description

In one specific edge case, mat-checkbox doesn't align with native checkbox. That's when checked and indeterminate are both set to true. On native checkbox, if indeterminate is true, then it renders indeterminate state (regardless of checked property). It also produced a sate of "mixed" in the a11y tree.

mat-checkbox handles it differently, since it only renders indeterminate state if the `checked` Input is `false`.

<mat-checkbox [checked]="true" [indeterminate]="true" />

### Reproduction

Steps to reproduce:
1. Create checkbox with code <mat-checkbox [checked]="true" [indeterminate]="true" />
2. Inspect a11y tree

### Expected Behavior

a11y tree has state of "mixed". This is what native checkbox does when the checked and indeterminate property (not attribute) are both set to true. Also, the aria state should match the visual state.

### Actual Behavior

Checkbox visually displays the indeterminate state, but a11y tree has state of "selected".

### Environment

- Angular:
- CDK/Material:
- Browser(s):
- Operating System (e.g. Windows, macOS, Ubuntu):

### PR DESCRIPTION
**Pull request by zarend**
For the checkbox component, fix semantics so that a11y produces a state of "mixed" when both the `checked` and `indeterminate` Inputs to mat-checkbox are both true.

Also, remove the use of aria-checked in favor of the native DOM properties for checked and indeterminate. Conform to [ARIA in HTML W3C input checkbox specification](https://www.w3.org/TR/html-aria/#el-input-checkbox).

Fix a11y bug where checkbox renders the indeterminate visual state but accessibility tree has a state of "checked". Aligns with native checkbox, which produces the mixed state when indeterminate property is true, regardless of the value of the checked property.

Some accessibility checkers produce an error when native checkbox has aria-checked on it. Remove aria-checked to fix errors.

Does not make visual changes.

Fix #26709

### CODE REVIEW

**Comment by zarend** [comment]
Note to reviewers, I confirmed that native checkbox always presents the indeterminate state when `indeterminate` DOM property is set to true. The aria state always matches the visual state, and native checkbox has the same behavior with indeterminate "true", checked "true" and indeterminate "true", checked "false". This looks to be on spec with HTML to me, and I confirmed that Chrome, Firefox and Safari all handle it the same way.


Visual and a11y tree state of native checkbox
|inputElement.checked|inputElement.indeterminate|visual state| a11y tree state|
|---|---|---|---|
|true|true|indeterminate|"mixed"|
|true|false|checked|"checked"|
|false|true|indeterminate|"mixed"|
|false|false|unchecked|"unchecked"|

File: src/material/checkbox/checkbox.ts [hunk 1]
```diff
@@ -312,11 +312,11 @@ export abstract class _MatCheckboxBase<E>
   }
 
   _getAriaChecked(): 'true' | 'false' | 'mixed' {
```
Comment (mmalerba) [inline src/material/checkbox/checkbox.ts:314]: can this function just be removed or is it still used somewhere?

Comment (zarend) [reply src/material/checkbox/checkbox.ts]: good catch. `_getAriaChecked` is not used anymore, and the legacy checkbox doesn't use it, so I deleted it.

**Review by mmalerba — APPROVED** [review]
LGTM

**Pull request merged**